coalib/bearlib/aspects/Smell.py

```diff
@@ -101,22 +98,84 @@ def func(a, b, c, d, e, f, g, h, i, j, k, l, m, n, o, p, q, r, s, t, z):
         parameters (maybe by creating more functions or using libraries).
         """
 
+
+@MethodSmell.subaspect
+class MethodLength:
+    """
+    Number of lines of code in a function or method definition.
+
+    Depending on the value of `method_length_count_comment`,
+    comments are considered.
+    """
+    class docs:
+        example = """
+        def is_positive(num):
+            if num >= 0:
+                return True
+            else:
+                return False
+
+        # This function can be defined in one line as follows:
+
+        def is_positive(num): return True if num >= 0 else False
+        """
+        example_language = 'python'
+        importance_reason = """
+        It is really important is to stay DRY (`Don't Repeat Yourself`) and
+        respect separation of concerns. Long methods are sometimes faster and
+        easier to write, and don't lead to maintenance problems. But most of
+        the time they are an easy way to detect that something *may* be wrong
+        in the code, and that special care is required while maintaining it.
+        """
+        fix_suggestions = """
+        Refactoring methods into smaller more generic methods, making code more
+        compact by using inline and language specific syntax tricks,
+        implementing methods or functions to avoid redundant operation(s),
+        making use of methods provided in libraries rather than reimplementing
+        them.
+        """
     max_method_length = Taste[int](
         'Represents the max number of lines for a method or a function\'s'
         'definition.',
-        (40,), default=40)
+        (10, 30, 60), default=30)
+    method_length_count_comment = Taste[bool](
+        'Allows when set to `True` to considered comments while calculating'
+        'methods\' length.',
+        (30, 60, 150), default=60)
+
+
+@MethodSmell.subaspect
+class ParameterListLength:
+    """
+    Number of parameter a function or method has.
+    """
+    class docs:
+        example = """
+        def func(a, b, c, d, e, f, g, h, i, j, k):
+            pass
+        """
+        example_language = 'python'
+        importance_reason = """
+        Methods that take too many parameters are difficult to read, maintain
+        and work with, and callers of those method often have an awkward time
+        assembling all of the data and the resulting code is usually not pretty.
+        """
+        fix_suggestions = """
+        This can be fixed by:
+
+        Instead of passing a group of data received from an object as
+        parameters, pass the object itself to the method.
+        Sometimes you can merge several parameters into a single object etc.
+        """
     max_parameters = Taste[int](
-        'Represents the max number of parameters for a function\'s.',
-        (5, 10), default=10)
+        'Represents the max number of parameters for a function or a method.',
+        (5, 7), default=7)
```

Comment:
Include that in the docstring. in the commit message, use "Code Complete, ISBN 0735619670" as that say it is a book, and is a permanent identifier, whereas bit.ly links can break.